### DISCUSSION TITLE
How to use rehype and remark plugin with next-mdx?

### DISCUSSION THREAD

**Discussion by Ajwad-Hameem-2007** [vercel/next.js#39135] [Help] [answered]
So, I was reading the docs of MDX and saw that I can use `rehype-highlight` with this. But when I installed it and added this code:

```js
// next.config.js

const withMDX = require('@next/mdx')({
    extension: /\.mdx?$/,
    options: {
      remarkPlugins: [],
      rehypePlugins: [rehypeHighlight],
      // If you use `MDXProvider`, uncomment the following line.
      providerImportSource: "@mdx-js/react",
    },
  })
  module.exports = withMDX({
    // Append the default value with md extensions
    pageExtensions: ['ts', 'tsx', 'js', 'jsx', 'md', 'mdx'],
  })
```

But it is not working. And I also tried to use `import rehypeHighlight from "rehype-highlight"` but it didn't work. Now can you guys tell me how to use remark and rehype plugins with mdx and next js?

**Comment by BelmaBesirevic** [comment]
I was wondering the same thing? Do we have to include these options in the getStaticProps function?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Could you please try as suggested here? https://github.com/vercel/next.js/discussions/36310

Reposting for visibility: 

> ```js
> // next.config.mjs
> import remarkGfm from "remark-gfm";
> import createMDX from "@next/mdx";
> 
> /** @type {import('next').NextConfig} */
> const nextConfig = {
>   reactStrictMode: true,
>   pageExtensions: ["js", "jsx", "ts", "tsx", "md", "mdx"],
> };
> 
> const withMDX = createMDX({
>   extension: /\.mdx?$/,
>   options: {
>     remarkPlugins: [remarkGfm],
>     rehypePlugins: [],
>   },
> });
> 
> export default withMDX(nextConfig);
> ```
